lib/util/mutex.js

```diff
@@ -0,0 +1,44 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.util.Mutex');
+
+
+/**
+ * @summary A simple mutex.
+ */
+shaka.util.Mutex = class {
+  /** Constructs the mutex. */
+  constructor() {
+    /** @private {boolean} */
+    this.acquired = false;
+
+    /** @private {!Array.<function()>} */
+    this.unlockQueue = [];
+  }
+
+  /**
+   * Acquires the mutex, as soon as possible.
+   * @return {!Promise}
+   */
+  async acquire() {
+    if (this.acquired) {
+      await new Promise((resolve) => this.unlockQueue.push(resolve));
```

Comment:
I think you need a way to timeout or cancel in the case of test failures.

Reply:
Done.